File: Leetcode/3403.py
class Solution:
    def answerString(self, word: str, numFriends: int) -> str:
        n = len(word)

        if n == 1 or numFriends == 1:
            return word

        if n == numFriends:
            return max(word)
        return max(word[i : i + (n - numFriends + 1)] for i in range(n))

File: Leetcode/3403.go
package Leetcode

func answerString(word string, numFriends int) string {
	n := len(word)

	if n == 1 || numFriends == 1 {
		return word
	}

	if numFriends == n {
		return maxCharacter(word)
	}

	ans := ""

	for i := range n {
		ans = max(ans, word[i:min(n, i+(n-numFriends+1))])
	}
	return ans
}

func maxCharacter(word string) string {
	ans := ""
	for _, ch := range word {
		ans = max(ans, string(ch))
	}
	return ans
}
